stage0/kvm.go

```diff
@@ -0,0 +1,119 @@
+// Copyright 2015 The rkt Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package stage0
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+)
+
+const (
+	kvmSettingsDir        = "/kvm"
+	kvmPrivateKeyFilename = "/ssh_kvm_key"
+)
+
+// fileAccessible checks if the given path exists and is a regular file
+func fileAccessible(path string) bool {
+	info, err := os.Stat(path)
+	if err == nil {
+		return info.Mode().IsRegular()
+	}
+	return false
+}
+
+func sshPrivateKeyPath(dataDir string) string {
+	return filepath.Join(dataDir, kvmSettingsDir, kvmPrivateKeyFilename)
+}
+
+func sshPublicKeyPath(dataDir string) string {
+	return filepath.Join(dataDir, kvmSettingsDir, kvmPrivateKeyFilename+".pub")
+}
+
+func createKvmSettingsDir(dataDir string) error {
+	return os.MkdirAll(filepath.Join(dataDir, kvmSettingsDir), 0700)
+}
+
+// generateKeyPair calls ssh-keygen with private key location for key generation purpose
+func generateKeyPair(private string) error {
+	out, err := exec.Command("ssh-keygen", "-q", "-t", "dsa", "-b", "1024", "-f", private).Output()
```

Comment:
during tests I was asked to "enter passphrase", please add 
`-N ""` to indicate no passphrase

additionally, consider making it more verbose like this:
```go
exec.Command("ssh-keygen",
    "-q", // silence
    "-t", "dsa", // type
    "-b", "1024", // bits
    "-f", private, // output file
    "-N", "", // no passphrase
)
```